cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/dns/HeadlessServiceDnsBootstrap.scala

```diff
@@ -279,16 +243,27 @@ final class HeadlessServiceDnsBootstrap(discovery: SimpleServiceDiscovery, setti
         case Some(contactPointProbingChild) ⇒
           Some(contactPointProbingChild)
         case None ⇒
-          val props = HttpContactPointBootstrap.props(settings, self, baseUri)
+          val props = HttpContactPointBootstrap.props(settings, contactPoint, baseUri)
           Some(context.actorOf(props, childActorName))
       }
   }
 
-  private def scheduleNextResolve(serviceName: String): Unit =
-    timers.startSingleTimer(TimerKeyResolveDNS, Internal.AttemptResolve(serviceName),
-      settings.contactPointDiscovery.interval)
+  private def decide(): Unit =
+    lastContactsObservation.foreach { contacts =>
+      val currentTime = timeNow()
+
+      // filter out old observations, in case the probing failures are not triggered
+      def isObsolte(obs: SeedNodesObservation): Boolean =
+        Duration.between(obs.observedAt, currentTime).toMillis > settings.contactPoint.probingFailureTimeout.toMillis
+
+      val seedObservations = seedNodesObservations.valuesIterator.filterNot(isObsolte).toSet
+      val info =
+        new SeedNodesInformation(currentTime, contacts.observedAt, contacts.observedContactPoints, seedObservations)
+
+      joinDecider.decide(info).recover { case _ => KeepProbing }.foreach(self ! _)
```

Comment:
recover is good, but also log a warning then?

Reply:
I guess no timeout is fine here hm; Technically we could end up invoking it again while it's still internally working on the previous decision future -- could lead to conflicting decisions? I guess we'd have acted upon the first thing already but maybe could show up as confusing logs if the decider decides to log things. 

Some more guard rails around this case would be nice... maybe if only logging that previous future decision still not completed etc